libsolidity/Parser.cpp

```diff
@@ -152,13 +153,21 @@ ASTPointer<ContractDefinition> Parser::parseContractDefinition()
 			VarDeclParserOptions options;
 			options.isStateVariable = true;
 			options.allowInitialValue = true;
+			options.isDeclaredConst = isDeclaredConst;
 			stateVariables.push_back(parseVariableDeclaration(options));
+			isDeclaredConst = false;
 			expectToken(Token::Semicolon);
 		}
 		else if (currentToken == Token::Modifier)
 			modifiers.push_back(parseModifierDefinition());
 		else if (currentToken == Token::Event)
 			events.push_back(parseEventDefinition());
+		else if (currentToken == Token::Const)
```

Comment:
Parsing the constant token should go inside parseVariableDeclaration (see below).